spec/p2p/p2p-top-level.md

```diff
@@ -0,0 +1,310 @@
+
+> mention other components and or specifications that are relevant for this
+spec. Possible interactions, possible use cases, etc. 
+
+> should give the reader the understanding in what environment this component
+will be used. 
+
+TODO:
+- Reactor API
+- Network? How do we communicate with other nodes
+- Discuss that validators run special set-up, and manage their own neighborhood (hide behind sentry nodes).
+   - As a result: the distributed system is composed of
+       - (correct) nodes that follow the protocol described here
+       - (potentially) adversarial nodes whose behavior deviates to harm the system
+       - (correct) nodes that don't follow the protocol to shield themselves but behave in a "nice way"
+
+
+
+# Part II - Sequential Definition of the  Problem
+
+
+##  Informal Problem statement
+
+
+The p2p layer, specified here, manages the connections of a Tendermint node with other Tendermint nodes. It continuously provides a list of peers ensuring
+1. Connectivity. The overlay network induced by the local neighborhoods (defined by the lists of peers) is sufficiently connected so that the reactors can implement communication on top of it that is sufficient for their needs
+    > There is the design decision that the same overlay is used by all reactors. It seems that consensus has the strongest requirements regarding connectivity and this defines the required properties
+2. Stability. Typically, connections between correct peers should be stable
+    > Even if at every time *t* we satisfy Point 1, if the overlays at times *t* and *t+1* are totally different, it might be hard to implement decent communication on top of it. E.g., Consensus gossip requires a neighbor to know its neighbors *k* state so that it can send the message to *k* that help *k* to advance. If *k* is connected only one second per hour, this is not feasible.
+3. Openness. It is always the case that new nodes can be added to the system
+    > Assuming 1. and 2. holds, this means, there must always be nodes that are willing to add connections to new peers.
+
+
+
+## Sequential Problem statement
+
+> should be English and precise. will be accompanied with a TLA spec.
+
+TODO: This seems to be a research question. Perhaps we can find some simple properties by looking at the peer-to-peer systems academic literature from several years ago?
+
+# Part III - Distributed System
+
+> Introduce distributed aspects 
+
+> Timing and correctness assumptions. Possibly with justification that the
+assumptions make sense, e.g., it is in the interest of a full node to behave
+correctly 
+
+> should have clear formalization in temporal logic.
+
+## Incentives
+
+TODO: 
+- who will follow the protocol who won't
+- validators hiding behind sentries (they have an incentive to not run it)
+- what can be incentives/strategies of bad nodes 
+     - DNS
+     - filling up all your connections and then disconnecting you
+     - feeding your reactors with garbage
+
```

Comment:
- isolating validators to prevent them from being coordinators, but using them to vote for proposals from the bad nodes

Reply:
added